libs/dal/src/repositories/layout/layout.repository.ts

```diff
@@ -149,4 +149,10 @@ export class LayoutRepository extends BaseRepository<EnforceEnvironmentQuery, La
 
     return updatedEntity;
   }
+
+  async findDeleted(_id: LayoutId, _environmentId: EnvironmentId): Promise<LayoutEntity> {
+    const res = await this.layout.findDeleted({ _id, _environmentId });
+
+    return this.mapEntity(res);
+  }
```

Comment:
@p-fernandez I was not sure how you would like to do the mapEntity. As I don't feel this deserves a usecase.

Reply:
Replied to this in the other comment regarding the use case. But the mapping as agreed in the last tech summit should live here, so that's ok. 👍🏻